src/loaders/yaml.cpp

```diff
@@ -163,10 +163,10 @@ bool loadFromYAMLNode(const YAML::Node& node, ReaderWriter& config, const Resolv
 #ifdef YAML_VERSION_0_3
             for(YAML::Iterator it2 = n.begin(); it2 != n.end(); ++it2)
 #else
-            for(YAML::const_iterator it2 = n.begin(); it2 != n.end(); ++it2)
+            for(std::size_t i = 0; i < n.size(); ++i)
```

Comment:
```suggestion
            for (const YAML::Node& n2 : n)
```

Reply:
I think this should work too and it more modern. Please check this interface isn't bugged too.

Reply:
I think this will have the bug too. According to https://stackoverflow.com/a/16504109/12135039, this uses the `begin` and `end` under the hood.